Homeworks/homework_oop_4/hw1.py

```diff
@@ -0,0 +1,94 @@
+"""
+Create classes to track homeworks.
+
+1. Homework - accepts howework text and deadline (datetime.timedelta)
+Homework has a method, that tells if deadline has passed.
+
+2. Student - can solve homework with `do_homework` method.
+Raises DeadlineError with "You are late" message if deadline has passed
+
+3. Teacher - can create homework with `create_homework`; check homework with `check_homework`.
+Any teacher can create or check any homework (even if it was created by one of colleagues).
+
+Homework are cached in dict-like structure named `homework_done`. Key is homework, values are
+solutions. Each student can only have one homework solution.
+
+Teacher can `reset_results` - with argument it will reset results for specific homework, without -
+it clears the cache.
+
+Homework is solved if solution has more than 5 symbols.
+
+
+Check file with tests to see how all these classes are used. You can create any additional classes
+you want.
+"""
+import datetime
+
+
+class Name:
+    def __init__(self, name, last_name):
+        self.name = name
+        self.last_name = last_name
+
+
+class Homework:
+    # arguments
+    def __init__(self, name_hw, deadline):
+        self.name_hw = name_hw
+        self.deadline = datetime.timedelta(deadline)
+
+    def is_expired(self):
+        if self.deadline <= datetime.timedelta(0):
+            return True
+        else:
+            return False
```

Comment:
https://docs.python.org/3/library/datetime.html#timedelta-objects
The deadline doesn't really do anything in your solution. It can look like this:
- when creating a hw, "remember" expiration date with by getting current datetime and adding timedelta with deadline
- when checking expiration date, check if current time is bigger then expiration date

Reply:
datetime.now() has been added